opcua/common/node.py

```diff
@@ -404,15 +404,9 @@ def read_event_history(self, starttime=None, endtime=None, numvalues=0, evtypes=
         if not type(evtypes) in (list, tuple):
             evtypes = [evtypes]
 
-        # FIXME not a very nice way to make sure events.get_filter gets a list of nodes...
-        evtype_nodes = []
-        for evtype in evtypes:
-            if not isinstance(evtype, Node):
-                evtype_nodes.append(Node(self.server, ua.NodeId(evtype)))  # make sure we have a list of Node objects
-            else:
-                evtype_nodes.append(evtype)
+        evtypes = [Node(self.server, evtype) for evtype in evtypes]
 
```

Comment:
Are you sure this will work? What if evtypes get passed a Node object? Some how I was having problems here if a Node object was passed, or maybe it was when only an int node ID was passed...